lib/kino_pythonx.ex

```diff
@@ -0,0 +1,212 @@
+            data = buffer.getvalue()
+            buffer.close()
+            data
+            """,
+            %{"value" => value}
+          )
+
+        data = Pythonx.decode(result)
+        Kino.Image.new(data, "image/png")
+      end)
+    end
+
+    Kino.Pythonx.register_render("altair.TopLevelMixin", fn value ->
+      {result, %{}} =
+        Pythonx.eval(
+          """
+          import json
+          # There is chart.to_json(), but we use chart.to_dict() to
+          # dump the JSON without whitespace.
+          json.dumps(value.to_dict(), separators=(',', ':'))
+          """,
+          %{"value" => value}
+        )
+
+      json = Pythonx.decode(result)
+      KinoPythonx.Altair.new(json)
+    end)
+  end
+end
+
+defimpl Kino.Render, for: Pythonx.Object do
+  def to_livebook(object) do
+    {classes, renders} = KinoPythonx.renders()
+
+    code = """
+    import sys
+
+    def render(value):
+      for (index, (mod_name, cls_name)) in enumerate(classes):
+        mod_name = mod_name.decode("utf-8")
+        cls_name = cls_name.decode("utf-8")
+        # If there is any object of this class, the module must have
+        # been imported, so we check for it in the imported modules,
+        # instead of trying to find and import it. This is important
+        # because importing a module for the first time can be time
+        # consuming, due to compilation.
+        if mod_name in sys.modules:
+          mod = sys.modules[mod_name]
+          cls = getattr(mod, cls_name)
+          if isinstance(value, cls):
+            return ("renderer", index)
+
+      if hasattr(value, "_repr_html_"):
+        html = value._repr_html_()
+        return ("html", html)
+
+      if hasattr(value, "_repr_png_"):
+        data = value._repr_png_()
+        return ("image", "png", data)
+
+      if hasattr(value, "_repr_jpeg_"):
+        data = value._repr_jpeg_()
+        return ("image", "jpeg", data)
+
+      if hasattr(value, "_repr_svg_"):
+        data = value._repr_svg_()
+        return ("image", "svg", data)
+
+      if hasattr(value, "_repr_markdown_"):
+        text = value._repr_markdown_()
+        return ("markdown", text)
+
+      if hasattr(value, "_repr_latex_"):
+        text = value._repr_latex_()
+        return ("latex", text)
+
+      return ("text", repr(value))
+
+    render(value)
+    """
+
+    {result, %{}} = Pythonx.eval(code, %{"value" => object, "classes" => classes})
+
+    kino =
+      case Pythonx.decode(result) do
+        {"text", text} -> Kino.Text.new(text, terminal: true)
+        {"html", html} -> Kino.HTML.new(html)
+        {"image", format, data} -> Kino.Image.new(data, "image/#{format}")
+        {"markdown", text} -> Kino.Markdown.new(text)
+        {"latex", text} -> Kino.Markdown.new("$$\n#{text}\n$$")
+        {"renderer", index} -> Enum.fetch!(renders, index).(object)
+      end
+
+    Kino.Render.to_livebook(kino)
+  end
+end
+
+defimpl Table.Reader, for: Pythonx.Object do
```

Comment:
This is the only one that I wonder if it should be part of Pythonx instead but glad to defer to your decision. :)

Reply:
I wondered the same, but given that it's about certain Python packages, and not Python itself, I am leaning towards doing it here, where we do other integrations :)